tests/kernel/sched/schedule_api/src/user_api.c

```diff
@@ -78,4 +86,272 @@ void test_user_k_is_preempt(void)
 
 	twp = thread_was_preempt;
 	zassert_true(twp, "unexpected return value");
+	k_thread_abort(tid);
+}
+
+/**
+ * userspace negative test: take NULL as input param to verify
+ * the api will trigger a fatal exception
+ */
+#ifdef CONFIG_USERSPACE
+static void thread_suspend_init_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+	k_thread_suspend(NULL);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_suspend() API
+ *
+ * @details Create a thread and set k_thread_suspend() input param to NULL
+ * will trigger a fatal error.
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see k_thread_suspend()
+ */
+void test_k_thread_suspend_init_null(void)
+{
+	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
+			(k_thread_entry_t)thread_suspend_init_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_k_thread_suspend_init_null(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+#ifdef CONFIG_USERSPACE
+static void thread_resume_init_null(void *p1, void *p2, void *p3)
+{
+	ztest_set_fault_valid(true);
+	k_thread_resume(NULL);
+
+	/* should not go here */
+	ztest_test_fail();
+}
+
+/**
+ * @brief Test k_thread_resume() API
+ *
+ * @details Create a thread and set k_thread_resume() input param to NULL
+ * will trigger a fatal error.
+ *
+ * @ingroup kernel_sched_tests
+ *
+ * @see k_thread_resume()
+ */
+void test_k_thread_resume_init_null(void)
+{
+	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
+			(k_thread_entry_t)thread_resume_init_null,
+			NULL, NULL, NULL,
+			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
+			K_USER | K_INHERIT_PERMS, K_NO_WAIT);
+
+	k_thread_join(tid, K_FOREVER);
+}
+#else
+void test_k_thread_resume_init_null(void)
+{
+	ztest_test_skip();
```

Comment:
Excuse me, last comment, could this be things like that? (for all of them)
```
void test_k_thread_resume_init_null(void)
{
#ifdef CONFIG_USERSPACE
	k_tid_t tid = k_thread_create(&user_thread, ustack, STACK_SIZE,
			(k_thread_entry_t)thread_resume_init_null,
			NULL, NULL, NULL,
			K_PRIO_PREEMPT(THREAD_TEST_PRIORITY),
			K_USER | K_INHERIT_PERMS, K_NO_WAIT);

	k_thread_join(tid, K_FOREVER);
#else
       ztest_test_skip();
#endif
}
```
I think it would be better when we generate doxygen documents.

Reply:
Got it, understand why you use it here.